File: src/test/resources/support-js/json_converter.js
var JsonObject = Packages.io.vertx.core.json.JsonObject;
var JsonArray = Packages.io.vertx.core.json.JsonArray;
module.exports = {
    toJsonObject: function(obj) {
        return new JsonObject(JSON.stringify(obj));
    },
    toJsonArray: function(arr) {
        return new JsonArray(JSON.stringify(arr));
    },
    fromJsonObject: function(obj) {
        return JSON.parse(obj.encode());
    },
    fromJsonArray: function(arr) {
        return JSON.parse(arr.encode());
    }
};

File: src/test/resources/support/json_converter.rb
require 'json'
java_import 'io.vertx.core.json.JsonObject'
java_import 'io.vertx.core.json.JsonArray'
module Support
  class JsonConverter
    def self.to_json_object obj
      return JsonObject.new(JSON.generate(obj))
    end
    def self.to_json_array arr
      return JsonArray.new(JSON.generate(arr))
    end
    def self.from_json_array arr
      return JSON.parse(arr.encode)
    end
  end
end
